var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -121,14 +120,15 @@ class Petsc(Package):
     depends_on('python@2.6:2.8', type='build')
 
     # Other dependencies
-    depends_on('boost', when='@:3.5+boost')
     depends_on('metis@5:~int64+real64', when='@:3.7.99+metis~int64+double')
     depends_on('metis@5:~int64', when='@:3.7.99+metis~int64~double')
     depends_on('metis@5:+int64+real64', when='@:3.7.99+metis+int64+double')
     depends_on('metis@5:+int64', when='@:3.7.99+metis+int64~double')
     # petsc-3.8+ uses default (float) metis with any (petsc) precision
     depends_on('metis@5:~int64', when='@3.8:+metis~int64')
     depends_on('metis@5:+int64', when='@3.8:+metis+int64')
+    # trilinos might require boost
+    depends_on('boost', when='@3.7.0:+trilinos+boost')
```

Comment:
If that's the case, shouldn't trilinos depend on boost, not petsc?

Reply:
Hm - ok - many these lines are not needed.

The important thing here is the next set of changes

```
         if 'trilinos' in spec:
             options.append('--with-cxx-dialect=C++11')
-            options.append('--with-boost=1')
+            if spec.satisfies('^trilinos+boost'):
+                options.append('--with-boost=1')
```

Will update PR